sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveShowCreateTableSuite.scala

```diff
@@ -61,129 +44,174 @@ class HiveShowCreateTableSuite extends ShowCreateTableSuite with TestHiveSinglet
+        SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "true") {
+      withTempDir { dir =>
+        withTable("t1") {
+          sql(
+            s"""CREATE TABLE t1 (
+               |  c1 INT COMMENT 'bla',
+               |  c2 STRING
+               |)
+               |LOCATION '${dir.toURI}'
+               |TBLPROPERTIES (
+               |  'prop1' = 'value1',
+               |  'prop2' = 'value2'
+               |)
+             """.stripMargin
+          )
+
+          checkCreateTable("t1")
+        }
+      }
+    }
+  }
+
   test("partitioned hive table") {
-    withTable("t1") {
-      sql(
-        s"""CREATE TABLE t1 (
-           |  c1 INT COMMENT 'bla',
-           |  c2 STRING
-           |)
-           |COMMENT 'bla'
-           |PARTITIONED BY (
-           |  p1 BIGINT COMMENT 'bla',
-           |  p2 STRING
-           |)
-         """.stripMargin
-      )
-
-      checkCreateTable("t1")
+    withSQLConf(
+        SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "true") {
+      withTable("t1") {
+        sql(
+          s"""CREATE TABLE t1 (
+             |  c1 INT COMMENT 'bla',
+             |  c2 STRING
+             |)
+             |COMMENT 'bla'
+             |PARTITIONED BY (
+             |  p1 BIGINT COMMENT 'bla',
+             |  p2 STRING
+             |)
+           """.stripMargin
+        )
+
+        checkCreateTable("t1")
+      }
     }
   }
 
   test("hive table with explicit storage info") {
-    withTable("t1") {
-      sql(
-        s"""CREATE TABLE t1 (
-           |  c1 INT COMMENT 'bla',
-           |  c2 STRING
-           |)
-           |ROW FORMAT DELIMITED FIELDS TERMINATED BY ','
-           |COLLECTION ITEMS TERMINATED BY '@'
-           |MAP KEYS TERMINATED BY '#'
-           |NULL DEFINED AS 'NaN'
-         """.stripMargin
-      )
-
-      checkCreateTable("t1")
+    withSQLConf(
+        SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "true") {
+      withTable("t1") {
+        sql(
+          s"""CREATE TABLE t1 (
+             |  c1 INT COMMENT 'bla',
+             |  c2 STRING
+             |)
+             |ROW FORMAT DELIMITED FIELDS TERMINATED BY ','
+             |COLLECTION ITEMS TERMINATED BY '@'
+             |MAP KEYS TERMINATED BY '#'
+             |NULL DEFINED AS 'NaN'
+           """.stripMargin
+        )
+
+        checkCreateTable("t1")
+      }
     }
   }
 
   test("hive table with STORED AS clause") {
-    withTable("t1") {
-      sql(
-        s"""CREATE TABLE t1 (
-           |  c1 INT COMMENT 'bla',
-           |  c2 STRING
-           |)
-           |STORED AS PARQUET
-         """.stripMargin
-      )
-
-      checkCreateTable("t1")
+    withSQLConf(
+        SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "true") {
+      withTable("t1") {
+        sql(
+          s"""CREATE TABLE t1 (
+             |  c1 INT COMMENT 'bla',
+             |  c2 STRING
+             |)
+             |STORED AS PARQUET
```

Comment:
@viirya, does this test also need to enable the configuration? I thought https://github.com/apache/spark/commit/58be82ad4b98fc17e821e916e69e77a6aa36209d changed the default provider when provider isn't specified